docs/ldml/tr35-personNames.md

```diff
@@ -41,26 +41,924 @@ The LDML specification is divided into the following parts:
+
+As an example for English, this may look like:
+
+```xml
+<personNames>
+    <personName order="givenFirst" length="long" usage="referring" formality="formal">
+<namePattern>{prefix} {given} {given2} {surname}, {suffix}</namePattern>
+    </personName>
+    <personName order="givenFirst" length="long" usage="referring" formality="informal">
+<namePattern>{given} «{given2}» {surname}</namePattern>
+<namePattern alt="2">«{given2}» {surname}</namePattern>
+    </personName>
+    <personName order="givenFirst" length="long" usage="sorting" formality="informal">
+<namePattern>{surname}, {given} {given2}</namePattern>
+    </personName>
+...
+</personNames>
+```
+
+The task is to find the best personName for a given set of input attributes. Well-formed data will always cover all possible combinations of the input parameters, so the algorithm is simple: traverse the list of person names until the first match is found, then return it.
+
+In more detail:
+
+A set of input parameters { order=O length=L usage=U formality=F } matches a personName element when:
+
+* The order attribute values contain O or there is no order attribute, and
+* The length attribute values contain L or there is no length attribute, and
+* The usage attribute values contain U or there is no usage attribute, and
+* The formality attribute values contain F or there is no formality attribute
+
+Example for input parameters 
+
+    order=**givenFirst** length=**long** usage=**referring** formality=**formal**
+
+To match a personName, all four attributes in the personName must match (a missing attribute matches any value for that attribute):
+
+| Sample personName attributes                                 | Matches? | Comment |
+| :----------------------------------------------------------- | :------: | :------ |
+| `order=_"givenFirst"_ length=_"long"_ usage=_"referring"_ formality=_"formal"_` | Y | exact match |
+| `length=_"long"_ usage=_"referring"_ formality=_"informal"_` | N | mismatch for formality |
+| `length=_"long"_ formality=_"formal"_`                       | Y | missing usage = all! |
+
+To find the matching personName element, traverse all the personNames in order until the first one is found. This will always terminate since the data is well-formed in CLDR.
+
+### 6.5 <a name="6-5-choose-a-namepattern" href="#6-5-choose-a-namepattern">Choose a namePattern</a>
+
+To format a name, the fields in a namePattern are replaced with the actual PersonName object to be formatted values from the input record. The personName element can contain multiple namePattern elements. We choose one based on the fields in the input PersonName object that are populated: The namePattern that has the most populated fields wins; if more than one have the same number of populated fields, the one with the least unpopulated fields wins. For example:
```

Comment:
"...are replaced with the actual PersonName object to be formatted values from the input record" reads really funny.  Maybe "...are replaced with the values of the named fields in the actual PersonNameObject"?

Reply:
How about: are replaced with fields fetched from the PersonName Data Interface.

(We should search for 'record' and fix those instances to either be name object or PN data interface)